homeassistant/components/crownstone/light.py

```diff
@@ -0,0 +1,204 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up crownstones from a config entry."""
+    manager: CrownstoneEntryManager = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities: list[CrownstoneEntity] = []
+
+    # Add Crownstone entities that support switching/dimming
+    for sphere in manager.cloud.cloud_data:
+        for crownstone in sphere.crownstones:
+            if crownstone.type in CROWNSTONE_INCLUDE_TYPES:
+                # Crownstone can communicate with Crownstone USB
+                if manager.uart and sphere.cloud_id == manager.usb_sphere_id:
+                    entities.append(CrownstoneEntity(crownstone, manager.uart))
+                # Crownstone can't communicate with Crownstone USB
+                else:
+                    entities.append(CrownstoneEntity(crownstone))
+
+    async_add_entities(entities)
+
+
+def crownstone_state_to_hass(value: int) -> int:
+    """Crownstone 0..100 to hass 0..255."""
+    return map_from_to(value, 0, 100, 0, 255)
+
+
+def hass_to_crownstone_state(value: int) -> int:
+    """Hass 0..255 to Crownstone 0..100."""
+    return map_from_to(value, 0, 255, 0, 100)
+
+
+class CrownstoneEntity(CrownstoneDevice, LightEntity):
+    """
+    Representation of a crownstone.
+
+    Light platform is used to support dimming.
+    """
+
+    _attr_should_poll = False
+    _attr_icon = "mdi:power-socket-de"
+
+    def __init__(self, crownstone_data: Crownstone, usb: CrownstoneUart = None) -> None:
+        """Initialize the crownstone."""
+        super().__init__(crownstone_data)
+        self.usb = usb
+        # Entity class attributes
+        self._attr_name = str(self.device.name)
+        self._attr_unique_id = f"{self.cloud_id}-{CROWNSTONE_SUFFIX}"
+
+    @property
+    def usb_available(self) -> bool:
+        """Return if this entity can use a usb dongle."""
+        return self.usb is not None and self.usb.is_ready()
+
+    @property
+    def brightness(self) -> int | None:
+        """Return the brightness if dimming enabled."""
+        return crownstone_state_to_hass(self.device.state)
+
+    @property
+    def is_on(self) -> bool:
+        """Return if the device is on."""
+        return crownstone_state_to_hass(self.device.state) > 0
+
+    @property
+    def supported_features(self) -> int:
+        """Return the supported features of this Crownstone."""
+        if self.device.abilities.get(DIMMING_ABILITY).is_enabled:
+            return SUPPORT_BRIGHTNESS
+        return 0
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """State attributes for Crownstone devices."""
+        attributes: dict[str, Any] = {}
+        # switch method
+        if self.usb_available:
+            attributes["switch_method"] = "Crownstone USB Dongle"
```

Comment:
Will this info change during the lifetime of the entity? If not it should not be part of the state attributes. Those are meant for dynamic info related to the state of the entity.

Reply:
Probably not since the entry it reloaded if the user decides to set up a USB dongle later, so i guess binary sensor too for this one? Though it would then fall out of the scope of this PR, since it only implemented the light platform. And then the only way for the user to check out that the dongle is indeed being used, are the logs.

Reply:
If we think this information is important to have in the state machine, ie to use in automations etc, we can use a binary sensor.